src/main/java/club/sk1er/hytilities/handlers/lobby/tab/TabChanger.java

```diff
@@ -74,4 +126,60 @@ private static boolean isSkyblockTabInformationEntry(NetworkPlayerInfo networkPl
             skyblockTabInformationEntryGameProfileNameRegex.matcher(networkPlayerInfo.getGameProfile().getName()).matches() &&
             !validMinecraftUsername.matcher(networkPlayerInfo.getDisplayName().getUnformattedText()).matches();
     }
+
+    /**
+     * List of usernames of Hypixel Friends.
+     * Generated by {@link #modifyName(String)}, which runs {@link #fetchFriendInfoThread}.
+     */
+    private static List<String> friends = null;
+
+    /**
+     * Thread that runs {@link #fetchFriendInfo()}.
+     * This thread will be created by {@link #modifyName(String)}.
+     * If the thread is null, then no such process is started.
+     */
+    private static Thread fetchFriendInfoThread = null;
+
+    /**
+     * Fetches the friend information for the current player
+     * This method is run by {@link #modifyName(String)} when it starts {@link #fetchFriendInfoThread}
+     *
+     * @return A list containing the friend list for a player, or null if the connection fails
+     */
+    private static List<String> fetchFriendInfo() {
+        try {
+            // Set URL to be https://api.sk1er.club/friends/(playerUUID)
+            URL url = new URL("https://api.sk1er.club/friends/" + Minecraft.getMinecraft().getSession().getPlayerID());
+
+            // Open connection with URL
+            HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+            connection.setRequestMethod("GET");
+            connection.setUseCaches(true);
+            connection.addRequestProperty("User-Agent", "Mozilla/4.76 (Hytilities Mod V1.0)");
+
+            // Connection will timeout after 15 seconds
+            connection.setReadTimeout(15000);
+            connection.setConnectTimeout(15000);
+
+            // Read output from connection
+            connection.setDoOutput(true);
+            InputStream is = connection.getInputStream();
```

Comment:
unclosed stream? can use `try (InputStream is = connection.getInputStream()) { // code }`, unless if this is done for another reason that i'm missing